hazelcast/src/main/java/com/hazelcast/jet/Job.java

```diff
@@ -349,4 +350,20 @@ default String getName() {
      * @throws UnsupportedOperationException if called for a light job
      */
     JobStateSnapshot exportSnapshot(String name);
+
+    /**
+     * Associates the given listener to this job.
+     * @return The registration id
+     *
+     * @since 5.3
```

Comment:
The feature doesn't work for light jobs, but we still allow registering listeners.
I'd prefer if we enable the feature for light jobs too. If that's not possible, this method should throw.